sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -1703,7 +1780,7 @@ private static BasicDocumentRevision getFullRevisionFromCurrentCursor(Cursor cur
             parent = cursor.getLong(7);
         } else if(cursor.columnType(7) == Cursor.FIELD_TYPE_NULL) {
         } else {
-            throw new IllegalArgumentException("Unexpected type: " + cursor.columnType(7));
+            throw new IllegalArgumentException("Unexpected type: " + cursor.columnType(7)); //TODO Should be straight Runtime Exception? Definitely not an argument
```

Comment:
Pretty sure Runtime is fine here. It's something that shouldn't happen I guess.

Any idea why the check for the column check? Does the schema change in a way that this might be true?